### DISCUSSION TITLE
File system routing

### DISCUSSION THREAD

**Discussion by redbar0n** [vikejs/vike#566] [Discussion]
Looking at https://vite-plugin-ssr.com/file-structure#domain-driven

```
/marketing/pages/index/index.page.js # Landing page (URL: `/`)
/marketing/pages/about/index.page.js # About page (URL: `/about`)
/marketing/pages/jobs/index.page.js # Jobs page (URL: `/jobs`)
```

I get a bit overwhelmed with so many index files. Is it easy to change to this, instead?

```
/marketing/pages/index/index.page.js # Landing page (URL: `/`)
/marketing/pages/about/about.page.js # About page (URL: `/about`)
/marketing/pages/jobs/jobs.page.js # Jobs page (URL: `/jobs`)
```

It would make a whole lot more sense in my brain (personally I'm used to the lovable conventions in Ruby on Rails). It's also easier in VS Code when having multiple tabs open. Perhaps vps could treat the file with the same name as the folder as the index file, like that. Another benefit would perhaps be that you could put pages directly in the `/pages/` folder like `/marketing/pages/jobs.page.js` if the page didn't have any other associated files or components. Or even maybe `/marketing/jobs.page.js`which would be really terse and neat.

**Comment by brillout** [comment]
See https://github.com/brillout/vite-plugin-ssr/issues/53#issuecomment-1356167413.

**Reply by redbar0n** [reply]
yes, a single config file would further enable this suggestion (since no other export files would be needed). It would also make the single config file the most convenient way to get started. If it has a default config (filesystem routing as per the convention I suggested), so user wouldn’t have to specify all the route paths (but could do it later to add customise).
